Neo4j.Driver/Neo4j.Driver/Experimental/FluentQueries/ExecutableQuery.cs

```diff
@@ -23,55 +23,77 @@
 
 namespace Neo4j.Driver.Experimental.FluentQueries;
 
-internal class ExecutableQuery : IExecutableQuery
+internal class ExecutableQuery<T> : IExecutableQuery<T>
 {
     private IInternalDriver _driver;
+    private Func<IAsyncEnumerable<IRecord>, ValueTask<T>> _streamProcessor;
     private Query _query;
     private QueryConfig _queryConfig;
 
-    public ExecutableQuery(IInternalDriver driver, string cypher)
+    public static ExecutableQuery<IReadOnlyList<IRecord>> GetDefault(IInternalDriver driver, string cypher)
     {
+        return new ExecutableQuery<IReadOnlyList<IRecord>>(new Query(cypher), driver, null, ToListAsync);
+    }
+
+    private ExecutableQuery(
+        Query query,
+        IInternalDriver driver,
+        QueryConfig queryConfig,
+        Func<IAsyncEnumerable<IRecord>, ValueTask<T>> streamProcessor)
+    {
+        _query = query;
         _driver = driver;
-        _query = new Query(cypher, new {});
+        _queryConfig = queryConfig;
+        _streamProcessor = streamProcessor;
+    }
+
+    private static async ValueTask<IReadOnlyList<T>> ToListAsync<T>(IAsyncEnumerable<T> enumerable)
+    {
+        var result = new List<T>();
+        await foreach (var item in enumerable)
+        {
+            result.Add(item);
+        }
+
+        return result;
     }
 
-    public IExecutableQuery WithConfig(QueryConfig config)
+    public IExecutableQuery<T> WithConfig(QueryConfig config)
```

Comment:
Did we make a decision on whether to move to a set of 'With' methods rather than a config object? Something we can add after release if we think it is a better way of doing it.

Reply:
Yeah, the config object is spread around quite a lot at the moment, but yes, we'll do it with "withers".